src/Illuminate/Database/Eloquent/Factories/RealTimeFactory.php

```diff
@@ -0,0 +1,520 @@
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use RuntimeException;
+
+class RealTimeFactory extends Factory
+{
+    use ManagesTypeMapping;
+
+    /**
+     * An instance of the factory's corresponding model.
+     *
+     * @var \Illuminate\Database\Eloquent\Model
+     */
+    protected $modelInstance;
+
+    /**
+     * The database schema manager for the model.
+     *
+     * @var \Doctrine\DBAL\Schema\AbstractSchemaManager
+     */
+    protected $schema;
+
+    /**
+     * The table name for the model.
+     *
+     * @var string
+     */
+    protected $table;
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return $this->configure()
+            ->getColumnsFromModel()
+            ->reject(fn (Column $column) => $column->getAutoincrement() || $this->isForeignKey($column) || $this->isPrimaryKey($column))
+            ->map(fn (Column $column) => $this->value($column))
+            ->all();
+    }
+
+    /**
+     * Configure the factory.
+     */
+    public function configure(): self
+    {
+        if (! interface_exists('Doctrine\DBAL\Driver')) {
+            throw new Exception('Real-time factories require the Doctrine DBAL (doctrine/dbal) package.');
+        }
+
+        $modelName = $this->modelName();
+        $this->modelInstance = new $modelName;
+        $connection = $this->modelInstance->getConnection();
+        $this->schema = $connection->getDoctrineSchemaManager();
+        $this->table = $this->modelInstance->getConnection()->getTablePrefix().$this->modelInstance->getTable();
+        $this->registerTypeMappings($connection->getDoctrineConnection()->getDatabasePlatform());
+
+        return $this;
+    }
+
+    /**
+     * Set the model for the factory.
+     */
+    public function forModel(string $model): self
+    {
+        $this->model = $model;
+
+        return $this;
+    }
+
+    /**
+     * Create a new instance of the factory builder with the given mutated properties.
+     *
+     * @return static
+     */
+    protected function newInstance(array $arguments = [])
+    {
+        return parent::newInstance($arguments)
+            ->forModel($this->model)
+            ->configure();
+    }
+
+    /**
+     * Get a new factory instance for the given attributes.
+     *
+     * @param  (callable(array<string, mixed>): array<string, mixed>)|array<string, mixed>  $attributes
+     * @return static
+     */
+    public static function new($attributes = [])
+    {
+        throw new RuntimeException('Real-time factories cannot be instantiated with new()');
+    }
+
+    /**
+     * Get the table columns from the model.
+     */
+    protected function getColumnsFromModel(): Collection
+    {
+        $columns = $this->schema->listTableColumns($this->table);
+
+        return collect($columns)->keyBy(fn ($column) => $column->getName());
```

Comment:
It seems possible to do without the `$columns` variable like in `isForeignKey()` or `isPrimaryKey()` in this class below:
```php
return collect($this->schema->listTableColumns($this->table))
       ->keyBy(fn ($column) => $column->getName());
```